contracts/simple_dex/lib.rs

```diff
@@ -82,8 +105,8 @@ mod simple_dex {
     pub struct SimpleDex {
         pub swap_fee_percentage: u128,
         pub access_control: AccountId,
-        // pool tokens
-        pub tokens: [AccountId; 4],
+        // a set of pairs that are availiable for swapping between
+        pub swap_pairs: Mapping<SwapPair, ()>,
```

Comment:
This is equivalent to `Mapping<AccountID, Vec<AccountID>>`, correct? 

I'm just wondering whether it wouldn't be cheaper to use the contract later on if instead of initializing a `SwapPair` we would do 
```rust
swap_pairs
    .get(&token_in)
    .map(|tokens_out| tokens_out.find(&token_out).is_some())
    .or_else(false)
```